mmdet/datasets/custom.py

```diff
@@ -100,6 +100,10 @@ def __init__(self,
                 'might cause errors if the path is not a local path. '
                 'Please use MMCV>= 1.3.16 if you meet errors.')
             self.data_infos = self.load_annotations(self.ann_file)
+        # assert an error when the data info is empty
+        assert len(self.data_infos) != 0, \
+            'empty data_infos, ' \
+            'which may be due to the \"image\" field of annotation json file'
```

Comment:
datasets like voc didn't use json file to save annotation. It's beter to replace it with `'Please check your dataset files.'`

Reply:
Thank you for your advice. I have updated it as you suggested.

Reply:
Adding this assert will cause some problem, unit test will fail https://github.com/open-mmlab/mmdetection/blob/7a9bc498d5cc972171ec4f7332afcd70bb50e60e/tests/test_data/test_datasets/test_custom_dataset.py#L21 .In this unit test, we use a empty dataset to test costum classes overriding. One possible solution: use a dataset which contains a sample instead of a empty dataset in unit test, the dataset can be put in `./test/data/`.